### PR TITLE
Rfc30/http

### PR DESCRIPTION
**Pull request by thomas-k-cameron**
## Motivation and Context
Part of RFC30.
Implements relevant logic for http/eventstream crate.

## Description
- add serde on external-type
- add serde-serialize/serde-deserialize

## Testing
Testing is not implemented yet.

## Checklist
CHANGELOG is updated once all the PR is approved.

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Comment by thomas-k-cameron** [comment]
@rcoh 
https://github.com/awslabs/smithy-rs/pull/1944#discussion_r1048766817

About this one, how do you think that it should handled?

File: rust-runtime/aws-smithy-eventstream/src/error.rs [hunk 1]
```diff
@@ -92,6 +99,10 @@ impl fmt::Display for Error {
             ),
             Marshalling(error) => write!(f, "failed to marshall message: {}", error),
             Unmarshalling(error) => write!(f, "failed to unmarshall message: {}", error),
+            DeserializedStream => write!(
+                f,
+                "this is a deserialized stream. No meesage can be sent or be received."
```
Comment (Velfi) [inline rust-runtime/aws-smithy-eventstream/src/error.rs:104]:
```suggestion
                "this is a deserialized stream. No message can be sent or be received."
```

**Comment by rcoh** [comment]
yeah I agree with @jdisanti — it's a two-way-door to add ser/de in the future. I think it's best to hold off on this and consider adding it at the end

**Comment by thomas-k-cameron** [comment]
I don't really have a strong opinion about event streams to be honest.

I looked up the crates.io but sdk that uses event-stream got less than 3000 all-time downloads, which makes it difficult to justify implementing something that is potentially harmful.

https://github.com/awslabs/smithy-rs/pull/2198#issuecomment-1397586646
You could create a new struct which copies it's twin without the field for the event stream, if there is a demand.

e.g.
```rust
struct OriginalStruct {
   event_stream: EventStreamType,
   other_field: u64
}

#[derive(serde::Serialize)]
struct Twin {
   other_field: u64
}
```

**Comment by thomas-k-cameron** [comment]
@Velfi 
@jdisanti 
@rcoh 

Can you guys check it out?
https://github.com/awslabs/smithy-rs/pull/2243

**Pull request closed without merging**